diff -purN a/process.c b/process.c
--- a/process.c	2009-03-06 02:25:10.000000000 +0100
+++ b/process.c	2011-01-16 15:01:51.000000000 +0100
@@ -373,8 +373,13 @@ int process_zipfiles(__G)    /* return P
     if (G.autorun_command[0] && !uO.qflag) { /* NO autorun without prompt! */
         Info(slide, 0x81, ((char *)slide, LoadFarString(AutorunPrompt),
                       FnFilter1(G.autorun_command)));
+#if defined WINDLL && defined WIN32
+        sprintf( (char*)slide, "Execute: %s", G.autorun_command);
+        if (IDOK == MessageBoxA(NULL, (LPCSTR)slide, "Auto-run command", MB_ICONQUESTION | MB_OKCANCEL))
+#else
         if (fgets(G.answerbuf, 9, stdin) != (char *)NULL
             && toupper(*G.answerbuf) == 'Y')
+#endif
             system(G.autorun_command);
         else
             Info(slide, 1, ((char *)slide, LoadFarString(NotAutoRunning)));
diff -purN a/windll/guisfx/sfxwiz.c b/windll/guisfx/sfxwiz.c
--- a/windll/guisfx/sfxwiz.c	2009-02-15 19:12:50.000000000 +0100
+++ b/windll/guisfx/sfxwiz.c	2011-01-16 15:03:14.000000000 +0100
@@ -342,6 +342,7 @@ BOOL FAR PASCAL InitDialogProc (HWND hDl
                getcwd(szTarget, _MAX_PATH);
 #endif
 
+               lpDCL->StructVersID = UZ_DCL_STRUCTVER;
                lpDCL->ncflag = 0;
                lpDCL->fQuiet = 0; // If not zero, no status messages will come through
                lpDCL->ntflag = 0;
@@ -394,6 +395,8 @@ BOOL FAR PASCAL InitDialogProc (HWND hDl
                      wsprintf(szMessage, "%s", "Error in archive");
                      break;
                 case PK_MEM:
+                     wsprintf(szMessage, "%s", "Insufficient memory to initialise");
+                     break;
                 case PK_MEM2:
                 case PK_MEM3:
                 case PK_MEM4:
@@ -422,11 +425,11 @@ BOOL FAR PASCAL InitDialogProc (HWND hDl
                      wsprintf(szMessage, "%s", "Unknown error");
                      break;
                 }
-               MessageBox(hDlg, szMessage, szAppName, MB_OK);
+//               MessageBox(hDlg, szMessage, szAppName, MB_OK);
 /* Uncomment line below to have SFXWix terminate automatically
    when done.
  */
-//              EndDialog(hDlg, wParam);
+              EndDialog(hDlg, wParam);
 
                break;
                case IDCANCEL:
