--- libdvdcss-1.2.10/configure.orig	2008-08-29 21:00:23.000000000 +0200
+++ libdvdcss-1.2.10/configure	2012-06-29 18:56:07.000000000 +0200
@@ -8303,7 +8303,7 @@ EOF
       # _LT_AC_TAGVAR(hardcode_libdir_flag_spec, ) is actually meaningless,
       # as there is no search path for DLLs.
       hardcode_libdir_flag_spec='-L$libdir'
-      allow_undefined_flag=unsupported
+#      allow_undefined_flag=unsupported
       always_export_symbols=no
       enable_shared_with_static_runtimes=yes
       export_symbols_cmds='$NM $libobjs $convenience | $global_symbol_pipe | $SED -e '\''/^[BCDGRS][ ]/s/.*[ ]\([^ ]*\)/\1 DATA/'\'' -e '\''/^[AITW][ ]/s/.*[ ]//'\'' | sort | uniq > $export_symbols'
@@ -8727,7 +8727,7 @@ if test -z "$aix_libpath"; then aix_libp
       # hardcode_libdir_flag_spec is actually meaningless, as there is
       # no search path for DLLs.
       hardcode_libdir_flag_spec=' '
-      allow_undefined_flag=unsupported
+#      allow_undefined_flag=unsupported
       # Tell ltmain to make .lib files, not .a files.
       libext=lib
       # Tell ltmain to make .dll files, not .so files.
