From 459960a7ca40819c51afa18b922c3c9f6390ca60 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Fri, 9 Dec 2011 12:48:50 +0100
Subject: [PATCH 083/285] eithelper: Add fixups for UK Freesat transponders

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythtv/eithelper.cpp |   37 ++++++++++++++++++++++++++++++++---
 1 file changed, 34 insertions(+), 3 deletions(-)

diff --git a/mythtv/libs/libmythtv/eithelper.cpp b/mythtv/libs/libmythtv/eithelper.cpp
index 299f0aa..4176bde 100644
--- a/mythtv/libs/libmythtv/eithelper.cpp
+++ b/mythtv/libs/libmythtv/eithelper.cpp
@@ -789,8 +789,9 @@ static void init_fixup(QMap<uint64_t,uint> &fix)
 
     // United Kingdom
     fix[ 9018U << 16] = EITFixUp::kFixUK;
-    // UK Freesat
+    // UK BBC
     fix[ 2013LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    // UK Freesat
     fix[ 2041LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2042LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2044LL << 32 | 2U << 16] = EITFixUp::kFixUK;
@@ -800,13 +801,43 @@ static void init_fixup(QMap<uint64_t,uint> &fix)
     fix[ 2048LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2049LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2050LL << 32 | 2U << 16] = EITFixUp::kFixUK;
-    fix[ 2051LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2053LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2054LL << 32 | 2U << 16] = EITFixUp::kFixUK;
-    fix[ 2056LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2057LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2063LL << 32 | 2U << 16] = EITFixUp::kFixUK;
     fix[ 2068LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    // R.caroline
+    fix[ 2315LL << 32 | 59U << 16] = EITFixUp::kFixUK;
+    // UK Sky
+    fix[ 2043LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2051LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2052LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2055LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2056LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2311LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2312LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2302LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2301LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2303LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2314LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2304LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2313LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2315LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2316LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2306LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    for (int i = 2402; i <= 2412; ++i)
+        fix[ (long long)i << 32 | 2U << 16] = EITFixUp::kFixUK;
+    for (int i = 2001; i <= 2012; ++i)
+       fix[ (long long)i << 32 | 2U << 16] = EITFixUp::kFixUK;
+    for (int i = 2014; i <= 2040; ++i)
+       fix[ (long long)i << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2611LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2612LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2601LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2602LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2613LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2603LL << 32 | 2U << 16] = EITFixUp::kFixUK;
+    fix[ 2604LL << 32 | 2U << 16] = EITFixUp::kFixUK;
 
     // ComHem Sweden
     fix[40999U << 16       ] = EITFixUp::kFixComHem;
-- 
1.7.9.5

