From f820e62311d9519190828f93951c940a28d6eaeb Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Thu, 30 Aug 2012 10:12:00 +0200
Subject: [PATCH 149/285] Mythgallery: Add .mts files to list of supported
 movies

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 .../mythgallery/mythgallery/galleryutil.cpp        |    1 +
 1 file changed, 1 insertion(+)

diff --git a/mythplugins/mythgallery/mythgallery/galleryutil.cpp b/mythplugins/mythgallery/mythgallery/galleryutil.cpp
index 6a838f2..0450bed 100644
--- a/mythplugins/mythgallery/mythgallery/galleryutil.cpp
+++ b/mythplugins/mythgallery/mythgallery/galleryutil.cpp
@@ -79,6 +79,7 @@ QStringList GalleryUtil::GetMovieFilter(void)
     filt.push_back("*.mp4");
     filt.push_back("*.mpeg");
     filt.push_back("*.mov");
+    filt.push_back("*.mts");
     filt.push_back("*.wmv");
     return filt;
 }
-- 
1.7.9.5

