From 02bfc9e167237f46924291620bd6406ab983bf3b Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Sat, 11 May 2013 15:51:54 +0100
Subject: [PATCH 185/285] netstream: Prevent memory leak on stream close

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythtv/netstream.cpp |   51 ++++++++++++++++++++---------------
 1 file changed, 30 insertions(+), 21 deletions(-)

diff --git a/mythtv/libs/libmythtv/netstream.cpp b/mythtv/libs/libmythtv/netstream.cpp
index 24d4a9c..1d7eb01 100644
--- a/mythtv/libs/libmythtv/netstream.cpp
+++ b/mythtv/libs/libmythtv/netstream.cpp
@@ -45,6 +45,7 @@ using std::getenv;
  */
 static QAtomicInt s_nRequest(1); // Unique NetStream request ID
 static QMutex s_mtx; // Guard local static data e.g. NAMThread singleton
+const qint64 kMaxBuffer = 4 * 1024 * 1024L; // 0= unlimited, 1MB => 4secs @ 1.5Mbps
 
 
 /*
@@ -167,7 +168,7 @@ bool NetStream::Request(const QUrl& url)
 
     if (m_reply)
     {
-        // Abort the current request
+        // Abort the current reply
         // NB the abort method appears to only work if called from NAMThread
         m_reply->disconnect(this);
         NAMThread::PostEvent(new NetStreamAbort(m_id, m_reply));
@@ -196,8 +197,9 @@ bool NetStream::Request(const QUrl& url)
 #endif
 #endif
 
-    VERBOSE(VB_FILE, LOC + QString("(%1) Requesting %2 from %3")
-        .arg(m_id).arg(m_request.url().toString()).arg(Source(m_request)) );
+    VERBOSE(VB_FILE, LOC + QString("(%1) Request %2 bytes=%3- from %4")
+        .arg(m_id).arg(m_request.url().toString())
+        .arg(m_pos).arg(Source(m_request)) );
     m_pending = new NetStreamRequest(m_id, m_request);
     NAMThread::PostEvent(m_pending);
     return true;
@@ -215,12 +217,13 @@ void NetStream::slotRequestStarted(int id, QNetworkReply *reply)
 
     if (!m_reply)
     {
-        VERBOSE(VB_FILE|VB_EXTRA, LOC + QString("(%1) Started %2-").arg(m_id).arg(m_pos) );
+        VERBOSE(VB_FILE, LOC + QString("(%1) Started 0x%2")
+            .arg(m_id).arg(quintptr(reply),0,16) );
 
         m_reply = reply;
         m_state = kStarted;
 
-        reply->setReadBufferSize(4*1024*1024L); // 0= unlimited, 1MB => 4secs @ 1.5Mbps
+        reply->setReadBufferSize(kMaxBuffer);
 
         // NB The following signals must be Qt::DirectConnection 'cos this slot
         // was connected Qt::DirectConnection so the current thread is NAMThread
@@ -294,23 +297,26 @@ void NetStream::slotReadyRead()
 
     if (m_reply)
     {
-        VERBOSE(VB_FILE|VB_EXTRA, LOC + QString("(%1) Ready %2 bytes")
-            .arg(m_id).arg(m_reply->bytesAvailable()) );
+        qint64 avail = m_reply->bytesAvailable();
+        VERBOSE((avail <= 2 * kMaxBuffer) ? VB_FILE|VB_EXTRA :
+                (avail <= 4 * kMaxBuffer) ? VB_FILE : VB_IMPORTANT,
+             LOC + QString("(%1) Ready 0x%2, %3 bytes available").arg(m_id)
+                .arg(quintptr(m_reply),0,16).arg(avail) );
 
-        if (m_size < 0)
+        if (m_size < 0 || m_state < kReady)
         {
             qlonglong first, last, len = ContentRange(m_reply, first, last);
             if (len >= 0)
             {
                 m_size = len;
-                VERBOSE(VB_FILE, LOC + QString("(%1) range %2-%3/%4")
-                    .arg(m_id).arg(first).arg(last).arg(len) );
+                VERBOSE(VB_FILE, LOC + QString("(%1) Ready 0x%2, range %3-%4/%5")
+                    .arg(m_id).arg(quintptr(m_reply),0,16).arg(first).arg(last).arg(len) );
             }
             else
             {
                 m_size = ContentLength(m_reply);
-                VERBOSE(VB_FILE, LOC + QString("(%1) content length %2")
-                    .arg(m_id).arg(m_size) );
+                VERBOSE(VB_FILE, LOC + QString("(%1) Ready 0x%2, content length %3")
+                    .arg(m_id).arg(quintptr(m_reply),0,16).arg(m_size) );
             }
         }
 
@@ -372,8 +378,8 @@ void NetStream::slotFinished()
 
         if (m_state == kFinished)
         {
-            VERBOSE(VB_FILE, LOC + QString("(%1) Finished %2/%3 bytes from %4")
-                .arg(m_id).arg(m_pos).arg(m_size).arg(Source(m_reply)) );
+            VERBOSE(VB_FILE, LOC + QString("(%1) Finished 0x%2 %3/%4 bytes from %5")
+                .arg(m_id).arg(quintptr(m_reply),0,16).arg(m_pos).arg(m_size).arg(Source(m_reply)) );
 
             locker.unlock();
             emit Finished(this);
@@ -454,9 +460,10 @@ void NetStream::Abort()
         m_pending = 0;
     }
 
-    if (m_reply && m_reply->isRunning())
+    if (m_reply)
     {
-        VERBOSE(VB_FILE, LOC + QString("(%1) Abort").arg(m_id) );
+        VERBOSE(VB_FILE, LOC + QString("(%1) Abort 0x%2")
+            .arg(m_id).arg(quintptr(m_reply),0,16) );
         NAMThread::PostEvent(new NetStreamAbort(m_id, m_reply));
         // NAMthread will delete the reply
         m_reply = 0;
@@ -670,7 +677,7 @@ NAMThread::~NAMThread()
 // virtual
 void NAMThread::run()
 {
-    VERBOSE(VB_MHEG, LOC "NAMThread starting");
+    VERBOSE(VB_GENERAL, LOC "NAMThread starting");
 
     m_nam = new QNetworkAccessManager();
     m_nam->setObjectName("NetStream NAM");
@@ -697,7 +704,7 @@ void NAMThread::run()
             QNetworkProxy::NoProxy;
         if (QNetworkProxy::NoProxy != type)
         {
-            VERBOSE(VB_MHEG, LOC "Using proxy: " + proxy);
+            VERBOSE(VB_GENERAL, LOC "Using proxy: " + proxy);
             m_nam->setProxy(QNetworkProxy(
                 type, url.host(), url.port(), url.userName(), url.password() ));
         }
@@ -733,7 +740,7 @@ void NAMThread::run()
     delete m_nam;
     m_nam = 0;
 
-    VERBOSE(VB_MHEG, LOC "NAMThread stopped");
+    VERBOSE(VB_GENERAL, LOC "NAMThread stopped");
 }
 
 // slot
@@ -775,8 +782,9 @@ bool NAMThread::StartRequest(NetStreamRequest *p)
 
     if (!p->m_bCancelled)
     {
-        VERBOSE(VB_FILE|VB_EXTRA, LOC + QString("(%1) StartRequest").arg(p->m_id) );
         QNetworkReply *reply = m_nam->get(p->m_req);
+        VERBOSE(VB_FILE|VB_EXTRA, LOC + QString("(%1) StartRequest 0x%2")
+            .arg(p->m_id).arg(quintptr(reply),0,16) );
         emit requestStarted(p->m_id, reply);
     }
     else
@@ -792,7 +800,8 @@ bool NAMThread::AbortRequest(NetStreamAbort *p)
         return false;
     }
 
-    VERBOSE(VB_FILE, LOC + QString("(%1) AbortRequest").arg(p->m_id) );
+    VERBOSE(VB_FILE|VB_EXTRA, LOC + QString("(%1) AbortRequest 0x%2").arg(p->m_id)
+        .arg(quintptr(p->m_reply),0,16) );
     p->m_reply->abort();
     p->m_reply->disconnect();
     delete p->m_reply;
-- 
1.7.9.5

