From 542c7c88638d4298abb1733bbe4b501b9fc935c2 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Sat, 5 Jan 2013 15:45:45 +0000
Subject: [PATCH 005/207] configure: Fix thread model selection for win32
 cross build

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/configure |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mythtv/configure b/mythtv/configure
index 8086821..b49e511 100755
--- a/mythtv/configure
+++ b/mythtv/configure
@@ -4908,12 +4908,12 @@ fi
 
 #End MythTV check
 
+enabled ffmpeg_pthreads && enable pthreads
+
 if ! disabled w32threads && ! enabled pthreads; then
     check_func_headers "windows.h process.h" _beginthreadex && enable w32threads
 fi
 
-enabled ffmpeg_pthreads && enable pthreads
-
 # check for some common methods of building with pthread support
 # do this before the optional library checks as some of them require pthreads
 if ! disabled pthreads && ! enabled w32threads; then
-- 
1.7.9.5

