From f965cd819a7d227c40f9c15d5b53f6e0a17f7ab5 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Fri, 1 Jun 2012 20:20:11 +0200
Subject: [PATCH 036/207] MHEG: Log text contents when updated

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythfreemheg/Text.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/mythtv/libs/libmythfreemheg/Text.cpp b/mythtv/libs/libmythfreemheg/Text.cpp
index 3c2e8fb..e4fc98b 100644
--- a/mythtv/libs/libmythfreemheg/Text.cpp
+++ b/mythtv/libs/libmythfreemheg/Text.cpp
@@ -357,6 +357,8 @@ void MHText::ContentPreparation(MHEngine *engine)
 void MHText::ContentArrived(const unsigned char *data, int length, MHEngine *engine)
 {
     CreateContent(data, length, engine);
+    MHLOG(MHLogDetail, QString("Content arrived %1 := '%2'")
+        .arg(m_ObjectReference.Printable()).arg(m_Content.Printable()) );
     // Now signal that the content is available.
     engine->EventTriggered(this, EventContentAvailable);
     m_fNeedsRedraw = true;
-- 
1.7.9.5

