From 4ee959b5ac2253d36523af11976ee37ccd46180a Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Wed, 7 Aug 2013 11:47:20 +0100
Subject: [PATCH 161/207] Settings: Add missing QString::arg

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmyth/settings.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmyth/settings.cpp b/mythtv/libs/libmyth/settings.cpp
index c0d5da7..bdb5e02 100644
--- a/mythtv/libs/libmyth/settings.cpp
+++ b/mythtv/libs/libmyth/settings.cpp
@@ -200,7 +200,7 @@ void SelectSetting::setValue(int which)
     if ((which >= ((int) values.size())) || (which < 0))
     {
         LOG(VB_GENERAL, LOG_ERR,
-                 QString("SelectSetting::setValue(): invalid index: %1 size: ")
+                 QString("SelectSetting::setValue(): invalid index: %1 size: %2")
                      .arg(which).arg(values.size()));
     }
     else
-- 
1.7.9.5

