diff --git a/mythtv/mythtv/programs/mythmtd/mythmtd.pro.orig b/mythtv/mythtv/programs/mythmtd/mythmtd.pro
index 4dc408b..f279293 100644
--- a/mythtv/mythtv/programs/mythmtd/mythmtd.pro.orig
+++ b/mythtv/mythtv/programs/mythmtd/mythmtd.pro
@@ -17,6 +17,7 @@ HEADERS += jobthread.h dvdprobe.h fileobs.h threadevents.h
 
 LIBS += -L../../libs/libmythdvdnav
 LIBS += -lmythdvdnav-$$LIBVERSION
+LIBS += -L../libmythtv              -lmythtv-$${LIBVERSION}
 
 SOURCES += main.cpp ../../libs/libmythtv/videodbcheck.cpp logging.cpp mtd.cpp
 SOURCES += jobthread.cpp dvdprobe.cpp fileobs.cpp threadevents.cpp
