From 84454bc77fb38e255ba879713a7a9c4741a40819 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Fri, 1 Jun 2012 20:19:07 +0200
Subject: [PATCH 034/227] MHEG: Log application launch

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythfreemheg/Engine.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mythtv/libs/libmythfreemheg/Engine.cpp b/mythtv/libs/libmythfreemheg/Engine.cpp
index 955889f..2158ea8 100644
--- a/mythtv/libs/libmythfreemheg/Engine.cpp
+++ b/mythtv/libs/libmythfreemheg/Engine.cpp
@@ -21,6 +21,7 @@
 
 #include <QStringList>
 #include <QRegExp>
+#include <QThread>
 
 #include "Engine.h"
 #include "ParseNode.h"
@@ -274,6 +275,7 @@ bool MHEngine::Launch(const MHObjectRef &target, bool fIsSpawn)
 
     if (target.m_GroupId.Size() == 0) return false; // No file name.
     QString csPath = GetPathName(target.m_GroupId); // Get path relative to root.
+    MHLOG(MHLogNotifications, "NOTE Launching " + csPath);
 
     // Check that the file exists before we commit to the transition.
     // This may block if we cannot be sure whether the object is present.
@@ -282,6 +284,7 @@ bool MHEngine::Launch(const MHObjectRef &target, bool fIsSpawn)
     {
         if (!m_fBooting)
             EngineEvent(2); // GroupIDRefError
+        MHLOG(MHLogWarning, "WARN Launch not found " + csPath);
         return false;
     }
 
@@ -356,6 +359,7 @@ bool MHEngine::Launch(const MHObjectRef &target, bool fIsSpawn)
         // Activate the application. ....
         CurrentApp()->Activation(this);
         m_fInTransition = false; // The transition is complete
+        MHLOG(MHLogNotifications, "NOTE Launch completed OK");
         return true;
     }
     catch (...)
-- 
1.7.9.5

