From 0367b2d127fd22c30af1290f1e99646515c6ebbb Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Wed, 6 Mar 2013 20:19:04 +0000
Subject: [PATCH 042/227] MHEG: Compatiblity with Freeview HD apps

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythfreemheg/Engine.cpp |   33 +++++++++++++++++++++++---------
 1 file changed, 24 insertions(+), 9 deletions(-)

diff --git a/mythtv/libs/libmythfreemheg/Engine.cpp b/mythtv/libs/libmythfreemheg/Engine.cpp
index 2158ea8..f7af37b 100644
--- a/mythtv/libs/libmythfreemheg/Engine.cpp
+++ b/mythtv/libs/libmythfreemheg/Engine.cpp
@@ -1343,22 +1343,37 @@ bool MHEngine::GetEngineSupport(const MHOctetString &feature)
         // The UK profile 1.06 seems a bit confused on this point.  It is not clear whether
         // we are supposed to return true for UKEngineProfile(2) or not.
         if (strings[1] == "2")
-        {
             return true;
-        }
-        else
-        {
-            return false;
-        }
+        if (strings[1] == "1")
+            return true;
+        // 'The Space' on Freeview checks this...
+        if (strings[1] == "PANT11001")
+            return true;
+        return false;
     }
 
     // InteractionChannelExtension.
     if (strings[0] == "ICProfile" || strings[0] == "ICP") {
-        if (strings.count() != 2) return false;
+        if (strings.count() < 2) return false;
         if (strings[1] == "0")
-            return true; // // InteractionChannelExtension.
+            return true; // InteractionChannelExtension.
         if (strings[1] == "1")
-            return false; // ICStreamingExtension.
+            return true; // ICStreamingExtension. This is a deliberate lie
+        return false;
+    }
+
+    if (strings[0] == "HDExtension" || strings[0] == "HDE") {
+        if (strings.count() < 2) return false;
+        if (strings[1] == "0")
+            return false; // HDVideoExtension.
+        if (strings[1] == "1")
+            return false; // HDGraphicsPlaneExtension
+        return false;
+    }
+    if (strings[0] == "HDGraphicsPlaneExtension" || strings[0] == "HDG") {
+        if (strings.count() < 2) return false;
+        if (strings[1] == "0")
+            return true; // HDGraphicsPlaneExtension
         return false;
     }
 
-- 
1.7.9.5

