From 71acebd90da1bdb154972c372a52bc298f421186 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Fri, 5 Jul 2013 21:08:45 +0100
Subject: [PATCH 099/227] MediaMon: Fix verbose string arg

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythbase/mythmedia.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythbase/mythmedia.cpp b/mythtv/libs/libmythbase/mythmedia.cpp
index 3b744d0..25a81e8 100644
--- a/mythtv/libs/libmythbase/mythmedia.cpp
+++ b/mythtv/libs/libmythbase/mythmedia.cpp
@@ -223,7 +223,7 @@ MythMediaType MythMediaDevice::DetectMediaType(void)
         }
         else
         {
-            LOG(VB_MEDIA, LOG_NOTICE, QString("DetectMediaType(this=0x%1) unknown file type %1")
+            LOG(VB_MEDIA, LOG_NOTICE, QString("DetectMediaType(this=0x%1) unknown file type %2")
                 .arg(quintptr(this),0,16).arg(it.key()));
         }
     }
-- 
1.7.9.5

