From 9d2019d25f0b06cbdf9311b71427078d90becf57 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Tue, 20 Aug 2013 18:44:31 +0100
Subject: [PATCH 206/227] AO: Fix string format args

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythtv/audioinputalsa.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/audioinputalsa.cpp b/mythtv/libs/libmythtv/audioinputalsa.cpp
index 53c5f0e..0aa3731 100644
--- a/mythtv/libs/libmythtv/audioinputalsa.cpp
+++ b/mythtv/libs/libmythtv/audioinputalsa.cpp
@@ -290,7 +290,7 @@ int AudioInputALSA::PcmRead(void* buf, uint nbytes)
     }
     if (nframes > 0)
         LOG(VB_AUDIO, LOG_ERR, LOC_DEV +
-            QString("short pcm read, %1 of %2 frames, retries %1")
+            QString("short pcm read, %1 of %2 frames, retries %3")
                 .arg(to_read - nframes).arg(to_read).arg(retries));
     return snd_pcm_frames_to_bytes(pcm_handle, to_read - nframes);
 }
-- 
1.7.9.5

