From 80da29b887a2a415b16cea50f4870bde61a6acab Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Sat, 27 Jul 2013 19:40:19 +0100
Subject: [PATCH 024/202] AFD: Audio only streams

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythtv/avformatdecoder.cpp |   13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/mythtv/libs/libmythtv/avformatdecoder.cpp b/mythtv/libs/libmythtv/avformatdecoder.cpp
index d35c29a..a48caa9 100644
--- a/mythtv/libs/libmythtv/avformatdecoder.cpp
+++ b/mythtv/libs/libmythtv/avformatdecoder.cpp
@@ -4698,6 +4698,9 @@ bool AvFormatDecoder::GetFrame(DecodeType decodetype)
                 av_init_packet(pkt);
             }
 
+            // NB av_read_frame will block until
+            // either a frame is read or an error occurs
+            // so MythPlayer::DecoderLoop will be unable to pause or stop
             int retval = 0;
             if (!ic || ((retval = ReadPacket(ic, pkt, storevideoframes)) < 0))
             {
@@ -4812,7 +4815,17 @@ bool AvFormatDecoder::GetFrame(DecodeType decodetype)
             case AVMEDIA_TYPE_AUDIO:
             {
                 if (!ProcessAudioPacket(curstream, pkt, decodetype))
+                {
                     have_err = true;
+                    if (!hasVideo)
+                    {
+                        LOG(VB_PLAYBACK, LOG_INFO, LOC +
+                            "GetFrame: exiting due to audio decode error");
+                        av_free_packet(pkt);
+                        delete pkt;
+                        return false;
+                    }
+                }
                 else
                     GenerateDummyVideoFrames();
                 break;
-- 
1.7.9.5

