From 1c4cb7d282bb5feca55202c8c8977e180e944b19 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Wed, 27 Jun 2012 19:28:55 +0200
Subject: [PATCH 113/202] libmyth: Initialise ProgramInfo::programflags in all
 ctors

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmyth/programinfo.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/mythtv/libs/libmyth/programinfo.cpp b/mythtv/libs/libmyth/programinfo.cpp
index 3557357..7505c63 100644
--- a/mythtv/libs/libmyth/programinfo.cpp
+++ b/mythtv/libs/libmyth/programinfo.cpp
@@ -277,6 +277,7 @@ ProgramInfo::ProgramInfo(const ProgramInfo &other) :
  */
 ProgramInfo::ProgramInfo(uint _chanid, const QDateTime &_recstartts) :
     chanid(0),
+    programflags(FL_NONE),
     positionMapDBReplacement(NULL)
 {
     LoadProgramFromRecorded(_chanid, _recstartts);
@@ -825,6 +826,7 @@ ProgramInfo::ProgramInfo(
  */
 ProgramInfo::ProgramInfo(const QString &_pathname) :
     chanid(0),
+    programflags(FL_NONE),
     positionMapDBReplacement(NULL)
 {
     if (_pathname.isEmpty())
-- 
1.7.9.5

