From e6a0d42a7f25a078a631f5b1d2eb1fa2f714ea9a Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Wed, 22 May 2013 17:34:28 +0100
Subject: [PATCH 123/202] TV: Verify indexes into livetvchain list

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythtv/livetvchain.cpp |   24 ++++++++++++++----------
 1 file changed, 14 insertions(+), 10 deletions(-)

diff --git a/mythtv/libs/libmythtv/livetvchain.cpp b/mythtv/libs/libmythtv/livetvchain.cpp
index 61fe991..37c5761 100644
--- a/mythtv/libs/libmythtv/livetvchain.cpp
+++ b/mythtv/libs/libmythtv/livetvchain.cpp
@@ -25,7 +25,7 @@ static inline void clear(LiveTVChainEntry &entry)
  */
 LiveTVChain::LiveTVChain() : ReferenceCounter("LiveTVChain"),
     m_id(""), m_maxpos(0), m_lock(QMutex::Recursive),
-    m_curpos(0), m_cur_chanid(0),
+    m_curpos(-1), m_cur_chanid(0),
     m_switchid(-1), m_jumppos(0)
 {
     clear(m_switchentry);
@@ -222,7 +222,9 @@ void LiveTVChain::ReloadAll(const QStringList &data)
                       "WHERE chainid = :CHAINID ORDER BY chainpos;");
         query.bindValue(":CHAINID", m_id);
 
-        if (query.exec() && query.isActive() && query.size() > 0)
+        if (!query.exec() || !query.isActive())
+            MythDB::DBError("Chain: ReloadAll", query);
+        else if (query.size() > 0)
         {
             while (query.next())
             {
@@ -349,11 +351,13 @@ int LiveTVChain::ProgramIsAt(const ProgramInfo &pginfo) const
 int LiveTVChain::GetLengthAtCurPos(void)
 {
     QMutexLocker lock(&m_lock);
-    LiveTVChainEntry entry;
 
-    entry = m_chain[m_curpos];
-    if (m_curpos == ((int)m_chain.count() - 1))
-        return entry.starttime.secsTo(MythDate::current());
+    if (m_curpos < 0 || m_curpos >= m_chain.count())
+        return 0;
+
+    LiveTVChainEntry entry = m_chain[m_curpos];
+    if (m_curpos + 1 == m_chain.count())
+        return entry.starttime.secsTo(QDateTime::currentDateTime());
     else
         return entry.starttime.secsTo(entry.endtime);
 }
@@ -376,7 +380,7 @@ void LiveTVChain::SetProgram(const ProgramInfo &pginfo)
 
 bool LiveTVChain::HasNext(void) const
 {
-    return ((int)m_chain.count() - 1 > m_curpos);
+    return m_chain.count() > m_curpos + 1;
 }
 
 void LiveTVChain::ClearSwitch(void)
@@ -603,11 +607,11 @@ static QString toString(const LiveTVChainEntry &v)
 QString LiveTVChain::toString() const
 {
     QMutexLocker lock(&m_lock);
-    QString ret = QString("LiveTVChain has %1 entries\n").arg(m_chain.size());
+    QString ret = QString("LiveTVChain has %1 entries").arg(m_chain.size());
     for (uint i = 0; i < (uint)m_chain.size(); i++)
     {
-        ret += (QString((i==(uint)m_curpos) ? "* " : "  ") +
-                ::toString(m_chain[i]) + "\n");
+        ret += ("\n\t" + QString((i==(uint)m_curpos) ? "* " : "  ") +
+                ::toString(m_chain[i]));
     }
     return ret;
 }
-- 
1.7.9.5

