From 6318acc5f380e31c6524b8f49bb2f50e831ef2c0 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Sun, 7 Jul 2013 10:17:50 +0100
Subject: [PATCH 144/202] TV: Reduce log verbosity of GeneratePreviewImage

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythtv/previewgeneratorqueue.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/libs/libmythtv/previewgeneratorqueue.cpp b/mythtv/libs/libmythtv/previewgeneratorqueue.cpp
index 6319cc4..d93379c 100644
--- a/mythtv/libs/libmythtv/previewgeneratorqueue.cpp
+++ b/mythtv/libs/libmythtv/previewgeneratorqueue.cpp
@@ -391,7 +391,7 @@ QString PreviewGeneratorQueue::GeneratePreviewImage(
     }
     else if (needs_gen)
     {
-        LOG(VB_PLAYBACK, LOG_INFO, LOC +
+        LOG(VB_PLAYBACK, LOG_DEBUG, LOC +
             QString("Not requesting preview for %1,"
                     "as it is already being generated")
                 .arg(pginfo.toString(ProgramInfo::kTitleSubtitle)));
-- 
1.7.9.5

