From 5f43361f4a050795cefb5f4d3ce2e4a958bb38c0 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Thu, 19 Sep 2013 18:49:24 +0100
Subject: [PATCH 182/202] Mytharchive: Fix set but not used warning

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 .../mytharchive/mytharchivehelper/pxsup2dast.c     |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/mythplugins/mytharchive/mytharchivehelper/pxsup2dast.c b/mythplugins/mytharchive/mytharchivehelper/pxsup2dast.c
index e9a5c7e..b8aa8f8 100644
--- a/mythplugins/mytharchive/mytharchivehelper/pxsup2dast.c
+++ b/mythplugins/mytharchive/mytharchivehelper/pxsup2dast.c
@@ -700,7 +700,7 @@ static void pxsubtitle(const char * supfile, FILE * ofh, eu8 palette[16][3],
                 int  prev;
                 int x1 = -1, x2 = -1, y1 = -1, y2 = -1;
                 int top_field = -1, bot_field = -1;
-                int start = 0, end = 0;
+                int end = 0;
                 int colcon_length;
                 eu8 this_palette[4][3];
                 boundstr_init(&bs, ctrl, size - pack);
@@ -721,7 +721,6 @@ static void pxsubtitle(const char * supfile, FILE * ofh, eu8 palette[16][3],
                             case 0x00:      /* force display: */
                                 continue;
                             case 0x01:      /* start date (read above) */
-                                start = date;
                                 continue;
                             case 0x02:      /* stop date (read above) */
                                 end = date;
@@ -883,7 +882,7 @@ int sup2dast(const char *supfile, const char *ifofile ,int delay_ms)
 {
     exc_try 
     {
-        int jc, i;
+        int i;
         eu8 yuvpalette[16][3], rgbpalette[16][3];
         char fnbuf[1024];
         char * p;
@@ -973,7 +972,6 @@ int sup2dast(const char *supfile, const char *ifofile ,int delay_ms)
             strcpy(&buf[i - 3], "xml");
             unlink(buf);
             rename(fnbuf, buf);
-            jc = 0; 
         }
         p[0] = '\0';
         printf("Output files reside in %s\n", fnbuf);
-- 
1.7.9.5

